src/cpp/session/modules/SessionObjectExplorer.cpp

```diff
@@ -193,6 +194,80 @@ void onDeferredInit(bool)
       LOG_ERROR(error);
 }
 
+void onDetectChanges(module_context::ChangeSource source)
+{
+   DROP_RECURSIVE_CALLS;
+
+   if (!core::thread::isMainThread())
+      return;
+
+   // unlikely that data will change outside of a REPL
+   if (source != module_context::ChangeSourceREPL) 
+      return;
+
+   Error error;
+   r::sexp::Protect rProtect;
+   SEXP envCache;
```

Comment:
Nit: initialize this to `R_NilValue` just to ensure it always has a valid R value.